### ISSUE TITLE
🗑️ Deprecation of `toNonZeroInt` extension function with warning level

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#477]
## 📝 Description

We would like to deprecate the `Number.toNonZeroInt()` extension function with a warning level for using the `NonZeroInt.Companion.create(Number)` function instead.

## 🔗 Dependencies

This issue is blocked by the following ones:
- [x] #476

## ✅ Checklist

- [ ] Deprecate the declaration with a warning level, refactor its usages and update the [unreleased changelog].
- [ ] Close this issue as completed and update tracking ones if present.

[unreleased changelog]: https://github.com/kotools/types/blob/main/CHANGELOG.md#unreleased

**Comment by LVMVRQUXL** [comment]
Another issue for this task will be created when ready.
